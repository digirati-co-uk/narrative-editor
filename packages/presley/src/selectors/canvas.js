import { createSelector, createStructuredSelector } from 'reselect';

export const currentCanvasId = state => state.canvas.current;
export const canvasList = state => state.canvas.list;
export const currentCanvas = createSelector(
  currentCanvasId,
  canvasList,
  (id, list) => (id ? list[id] : null)
);

export const getLabel = createSelector(currentCanvas, canvas => canvas.label);
export const getSummary = createSelector(
  currentCanvas,
  canvas => canvas.summary
);
export const getMetadata = createSelector(
  currentCanvas,
  canvas => canvas.metadata
);
export const getRequiredStatement = createSelector(
  currentCanvas,
  canvas => canvas.requiredStatement
);
export const getId = createSelector(currentCanvas, canvas => canvas.id);
