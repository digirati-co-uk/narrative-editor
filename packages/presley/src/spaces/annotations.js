import { createActions, handleActions } from 'redux-actions';
import update from 'immutability-helper';
import { reorderArray } from '../utility';

// Constants
const ADD_ANNOTATION = 'ADD_ANNOTATION';
const UPDATE_ANNOTATION = 'UPDATE_ANNOTATION';
const UPDATE_ANNOTATION_ORDER = 'UPDATE_ANNOTATION_ORDER';
const REMOVE_ANNOTATION = 'REMOVE_ANNOTATION';
const RECOVER_ANNOTATION = 'RECOVER_ANNOTATION';

const DEFAULT_STATE = {
  order: [],
  list: {},
};

const {
  addAnnotation,
  updateAnnotation,
  updateAnnotationOrder,
  removeAnnotation,
  recoverAnnotation,
} = createActions({
  [ADD_ANNOTATION]: (id, annotation) => ({ id, annotation }),
  [UPDATE_ANNOTATION]: (id, annotation) => ({ id, annotation }),
  [UPDATE_ANNOTATION_ORDER]: (from, to) => ({ from, to }),
  [REMOVE_ANNOTATION]: annotationId => ({ annotationId }),
  [RECOVER_ANNOTATION]: annotationId => ({ annotationId }),
});

const reducer = handleActions(
  {
    [addAnnotation]: (state, { payload: { id, annotation } }) =>
      update(state, {
        order: { $push: [id] },
        list: {
          [id]: { $set: annotation },
        },
      }),

    [updateAnnotation]: (state, { payload: { id, annotation } }) =>
      update(state, {
        list: {
          [id]: { $set: annotation },
        },
      }),

    [updateAnnotationOrder]: (state, { payload: { from, to } }) =>
      update(state, {
        order: reorderArray(from, to),
      }),

    [removeAnnotation]: (state, { payload: { annotationId } }) =>
      update(state, {
        order: ids => ids.filter(id => id !== annotationId),
      }),

    [recoverAnnotation]: (state, { payload: { annotationId } }) =>
      state.list[annotationId]
        ? update(state, {
            order: { $push: [annotationId] },
          })
        : state,
  },
  DEFAULT_STATE
);

export {
  addAnnotation,
  updateAnnotation,
  updateAnnotationOrder,
  removeAnnotation,
  recoverAnnotation,
  reducer,
  DEFAULT_STATE,
};
