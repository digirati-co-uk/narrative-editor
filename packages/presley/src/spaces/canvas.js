// Constants.
import { createActions, handleActions } from 'redux-actions';
import { reorderArray } from '../utility';
import update from 'immutability-helper';

const CREATE_CANVAS = 'CREATE_CANVAS';
const REMOVE_CANVAS = 'REMOVE_CANVAS';
const RECOVER_CANVAS = 'RECOVER_CANVAS';
const UPDATE_CANVAS_ORDER = 'UPDATE_CANVAS_ORDER';
const SELECT_CANVAS = 'SELECT_CANVAS';

const CANVAS_UPDATE_LABEL = 'CANVAS_UPDATE_LABEL';
const CANVAS_UPDATE_SUMMARY = 'CANVAS_UPDATE_SUMMARY';
const CANVAS_ADD_METADATA_PAIR = 'CANVAS_ADD_METADATA_PAIR';
const CANVAS_REMOVE_METADATA_PAIR = 'CANVAS_REMOVE_METADATA_PAIR';
const CANVAS_UPDATE_METADATA_PAIR = 'CANVAS_UPDATE_METADATA_PAIR';
const CANVAS_REORDER_METADATA_PAIR = 'CANVAS_REORDER_METADATA_PAIR';
const CANVAS_UPDATE_REQUIRED_STATEMENT = 'CANVAS_UPDATE_REQUIRED_STATEMENT';

const DEFAULT_STATE = {
  current: null,
  list: {},
  order: [],
};

const DEFAULT_CANVAS_STATE = {
  label: null,
  summary: null,
  metadata: [],
  requiredStatement: null,
};

const {
  createCanvas,
  removeCanvas,
  recoverCanvas,
  updateCanvasOrder,
  selectCanvas,
  canvasUpdateLabel,
  canvasUpdateSummary,
  canvasAddMetadataPair,
  canvasUpdateMetadataPair,
  canvasRemoveMetadataPair,
  canvasReorderMetadataPair,
  canvasUpdateRequiredStatement,
} = createActions({
  [CREATE_CANVAS]: id => ({ id }),
  [REMOVE_CANVAS]: id => ({ id }),
  [RECOVER_CANVAS]: id => ({ id }),
  [SELECT_CANVAS]: id => ({ id }),
  [UPDATE_CANVAS_ORDER]: (from, to) => ({ from, to }),
  [CANVAS_UPDATE_LABEL]: (id, label) => ({ id, label }),
  [CANVAS_UPDATE_SUMMARY]: (id, summary) => ({ id, summary }),
  [CANVAS_ADD_METADATA_PAIR]: (id, label, value) => ({ id, label, value }),
  [CANVAS_REMOVE_METADATA_PAIR]: (id, index) => ({ id, index }),
  [CANVAS_UPDATE_METADATA_PAIR]: (id, index, label, value) => ({
    id,
    index,
    label,
    value,
  }),
  [CANVAS_REORDER_METADATA_PAIR]: (id, from, to) => ({ id, from, to }),
  [CANVAS_UPDATE_REQUIRED_STATEMENT]: (id, requiredStatement) => ({
    id,
    requiredStatement,
  }),
});

const reducer = handleActions(
  {
    [createCanvas]: (state, { payload: { id } }) =>
      update(state, {
        current: { $set: id },
        order: { $push: [id] },
        list: {
          [id]: { $set: { id, ...DEFAULT_CANVAS_STATE } },
        },
      }),

    [removeCanvas]: (state, { payload: { id } }) =>
      update(state, {
        current: { $set: state.current === id ? null : state.current },
        order: ids => ids.filter(currentId => currentId !== id),
      }),

    [recoverCanvas]: (state, { payload: { id } }) =>
      state.list[id]
        ? update(state, {
            order: { $push: [id] },
          })
        : state,

    [updateCanvasOrder]: (state, { payload: { from, to } }) =>
      update(state, {
        order: reorderArray(from, to),
      }),

    [selectCanvas]: (state, { payload: { id } }) =>
      update(state, {
        current: { $set: id },
      }),

    [canvasUpdateLabel]: (state, { payload: { id, label } }) =>
      update(state, {
        list: {
          [id]: {
            label: { $set: label },
          },
        },
      }),

    [canvasUpdateSummary]: (state, { payload: { id, summary } }) =>
      update(state, {
        list: {
          [id]: { summary: { $set: summary } },
        },
      }),

    [canvasAddMetadataPair]: (state, { payload: { id, label, value } }) =>
      update(state, {
        list: {
          [id]: {
            metadata: { $push: [{ label, value }] },
          },
        },
      }),

    [canvasUpdateMetadataPair]: (
      state,
      { payload: { id, index, label, value } }
    ) =>
      update(state, {
        list: {
          [id]: {
            metadata: metadata =>
              metadata.map(
                (mValue, mIndex) =>
                  index === mIndex ? { label, value } : mValue
              ),
          },
        },
      }),

    [canvasRemoveMetadataPair]: (state, { payload: { id, index } }) =>
      update(state, {
        list: {
          [id]: {
            metadata: metadata =>
              metadata.filter((value, key) => key !== index),
          },
        },
      }),

    [canvasReorderMetadataPair]: (state, { payload: { id, from, to } }) =>
      update(state, {
        list: {
          [id]: {
            metadata: reorderArray(from, to),
          },
        },
      }),

    [canvasUpdateRequiredStatement]: (
      state,
      { payload: { id, requiredStatement } }
    ) =>
      update(state, {
        list: {
          [id]: {
            requiredStatement: { $set: requiredStatement },
          },
        },
      }),
  },
  DEFAULT_STATE
);

export {
  createCanvas,
  removeCanvas,
  recoverCanvas,
  updateCanvasOrder,
  selectCanvas,
  canvasUpdateLabel,
  canvasUpdateSummary,
  canvasAddMetadataPair,
  canvasUpdateMetadataPair,
  canvasRemoveMetadataPair,
  canvasReorderMetadataPair,
  canvasUpdateRequiredStatement,
  reducer,
  DEFAULT_STATE,
  DEFAULT_CANVAS_STATE,
};
