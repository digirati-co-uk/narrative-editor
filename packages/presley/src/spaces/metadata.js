import { createActions, handleActions } from 'redux-actions';
import update from 'immutability-helper';
import { reorderArray } from '../utility';

// Constants
const UPDATE_LABEL = 'UPDATE_LABEL';
const UPDATE_SUMMARY = 'UPDATE_SUMMARY';
const ADD_METADATA_PAIR = 'ADD_METADATA_PAIR';
const REMOVE_METADATA_PAIR = 'REMOVE_METADATA_PAIR';
const UPDATE_METADATA_PAIR = 'UPDATE_METADATA_PAIR';
const REORDER_METADATA_PAIR = 'REORDER_METADATA_PAIR';
const UPDATE_REQUIRED_STATEMENT = 'UPDATE_REQUIRED_STATEMENT';

const DEFAULT_STATE = {
  label: null,
  summary: null,
  metadata: [],
  requiredStatement: null,
};

const {
  updateLabel,
  updateSummary,
  addMetadataPair,
  updateMetadataPair,
  removeMetadataPair,
  reorderMetadataPair,
  updateRequiredStatement,
} = createActions({
  [UPDATE_LABEL]: label => ({ label }),
  [UPDATE_SUMMARY]: summary => ({ summary }),
  [ADD_METADATA_PAIR]: (label, value) => ({ label, value }),
  [REMOVE_METADATA_PAIR]: index => ({ index }),
  [UPDATE_METADATA_PAIR]: (index, label, value) => ({ index, label, value }),
  [REORDER_METADATA_PAIR]: (from, to) => ({ from, to }),
  [UPDATE_REQUIRED_STATEMENT]: requiredStatement => ({ requiredStatement }),
});

const reducer = handleActions(
  {
    [updateLabel]: (state, { payload: { label } }) =>
      update(state, { label: { $set: label } }),

    [updateSummary]: (state, { payload: { summary } }) =>
      update(state, {
        summary: { $set: summary },
      }),

    [addMetadataPair]: (state, { payload: { label, value } }) =>
      update(state, {
        metadata: { $push: [{ label, value }] },
      }),

    [updateMetadataPair]: (state, { payload: { index, label, value } }) =>
      update(state, {
        metadata: metadata =>
          metadata.map(
            (mValue, mIndex) => (index === mIndex ? { label, value } : mValue)
          ),
      }),

    [removeMetadataPair]: (state, { payload: { index } }) =>
      update(state, {
        metadata: metadata => metadata.filter((value, key) => key !== index),
      }),

    [reorderMetadataPair]: (state, { payload: { from, to } }) =>
      update(state, {
        metadata: reorderArray(from, to),
      }),

    [updateRequiredStatement]: (state, { payload: { requiredStatement } }) =>
      update(state, {
        requiredStatement: { $set: requiredStatement },
      }),
  },
  DEFAULT_STATE
);

export {
  updateLabel,
  updateSummary,
  addMetadataPair,
  updateMetadataPair,
  removeMetadataPair,
  reorderMetadataPair,
  updateRequiredStatement,
  reducer,
  DEFAULT_STATE,
};
