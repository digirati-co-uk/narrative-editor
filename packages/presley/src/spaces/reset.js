export const RESET = '__GLOBAL_RESET__';

export function reset() {
  return { type: RESET };
}
