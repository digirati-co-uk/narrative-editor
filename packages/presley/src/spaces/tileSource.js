/**
 * Note: This only supports one tile source at the moment.
 */
import { createActions, handleActions } from 'redux-actions';
import update from 'immutability-helper';

// Constants.
const CHANGE_TILE_SOURCE = 'CHANGE_TILE_SOURCE';
const REMOVE_TILE_SOURCE = 'REMOVE_TILE_SOURCE';

const DEFAULT_STATE = {
  current: null,
};

const { changeTileSource, removeTileSource } = createActions({
  [CHANGE_TILE_SOURCE]: tileSource => ({ tileSource }),
  [REMOVE_TILE_SOURCE]: () => ({}),
});

const reducer = handleActions(
  {
    [changeTileSource]: (state, { payload: { tileSource } }) =>
      update(state, { current: { $set: tileSource } }),

    [removeTileSource]: state => update(state, { current: { $set: null } }),
  },
  DEFAULT_STATE
);

export { changeTileSource, removeTileSource, reducer, DEFAULT_STATE };
