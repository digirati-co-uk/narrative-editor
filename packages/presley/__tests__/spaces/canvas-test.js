import * as canvas from '../../src/spaces/canvas';
import { createStore, combineReducers } from 'redux';

describe('spaces/canvas', () => {
  let store;
  beforeEach(() => {
    store = createStore(combineReducers({ canvas: canvas.reducer }));
  });

  test('canvas can be created', () => {
    store.dispatch(canvas.createCanvas('id-123'));

    expect(store.getState()).toMatchSnapshot();
  });

  test('multiple canvases can be created', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.createCanvas('id-2'));

    expect(store.getState()).toMatchSnapshot();
  });

  test('canvas can be removed', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.createCanvas('id-2'));
    store.dispatch(canvas.createCanvas('id-3'));

    store.dispatch(canvas.removeCanvas('id-2'));

    expect(store.getState()).toMatchSnapshot();
  });

  test('canvas can be recovered', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.createCanvas('id-2'));
    store.dispatch(canvas.createCanvas('id-3'));

    store.dispatch(canvas.removeCanvas('id-2'));
    store.dispatch(canvas.recoverCanvas('id-2'));

    expect(store.getState()).toMatchSnapshot();
  });

  test('canvas that does not exist cannot be recovered', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.removeCanvas('id-1'));

    store.dispatch(canvas.recoverCanvas('id-2'));

    expect(store.getState()).toMatchSnapshot();
  });

  test('current canvas can be removed', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.removeCanvas('id-1'));

    expect(store.getState().canvas.current).toEqual(null);
  });

  test('canvas can be selected manually', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.createCanvas('id-2'));
    store.dispatch(canvas.createCanvas('id-3'));
    store.dispatch(canvas.selectCanvas('id-2'));

    expect(store.getState().canvas.current).toEqual('id-2');
  });

  test('canvas order can be updated', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.createCanvas('id-2'));
    store.dispatch(canvas.createCanvas('id-3'));

    store.dispatch(canvas.updateCanvasOrder(2, 1));

    expect(store.getState().canvas.order).toEqual(['id-1', 'id-3', 'id-2']);
  });

  test('canvas label can be updated', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.canvasUpdateLabel('id-1', 'Some test label'));

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: 'Some test label',
            metadata: [],
            requiredStatement: null,
            summary: null,
          },
        },
        order: ['id-1'],
      },
    });
  });

  test('canvas summary can be updated', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(canvas.canvasUpdateSummary('id-1', 'Test summary'));

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: null,
            metadata: [],
            requiredStatement: null,
            summary: 'Test summary',
          },
        },
        order: ['id-1'],
      },
    });
  });

  test('canvas metadata pair can be added', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label', 'test value')
    );

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: null,
            metadata: [{ label: 'Test label', value: 'test value' }],
            requiredStatement: null,
            summary: null,
          },
        },
        order: ['id-1'],
      },
    });
  });

  test('canvas metadata pair can be removed', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 1', 'test value 1')
    );
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 2', 'test value 2')
    );
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 3', 'test value 3')
    );
    store.dispatch(canvas.canvasRemoveMetadataPair('id-1', 1));

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: null,
            metadata: [
              { label: 'Test label 1', value: 'test value 1' },
              { label: 'Test label 3', value: 'test value 3' },
            ],
            requiredStatement: null,
            summary: null,
          },
        },
        order: ['id-1'],
      },
    });
  });

  test('canvas metadata pair can be updated', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 1', 'test value 1')
    );
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 2', 'test value 2')
    );
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 3', 'test value 3')
    );
    store.dispatch(
      canvas.canvasUpdateMetadataPair(
        'id-1',
        1,
        'Test label 2 change',
        'Test value 2 changed'
      )
    );

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: null,
            metadata: [
              { label: 'Test label 1', value: 'test value 1' },
              { label: 'Test label 2 change', value: 'Test value 2 changed' },
              { label: 'Test label 3', value: 'test value 3' },
            ],
            requiredStatement: null,
            summary: null,
          },
        },
        order: ['id-1'],
      },
    });
  });

  test('canvas metadata pair order can be updated', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 1', 'test value 1')
    );
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 2', 'test value 2')
    );
    store.dispatch(
      canvas.canvasAddMetadataPair('id-1', 'Test label 3', 'test value 3')
    );
    store.dispatch(canvas.canvasReorderMetadataPair('id-1', 1, 0));

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: null,
            metadata: [
              { label: 'Test label 2', value: 'test value 2' },
              { label: 'Test label 1', value: 'test value 1' },
              { label: 'Test label 3', value: 'test value 3' },
            ],
            requiredStatement: null,
            summary: null,
          },
        },
        order: ['id-1'],
      },
    });
  });

  test('canvas required statement can be updated', () => {
    store.dispatch(canvas.createCanvas('id-1'));
    store.dispatch(
      canvas.canvasUpdateRequiredStatement('id-1', 'Some required statement')
    );

    expect(store.getState()).toEqual({
      canvas: {
        current: 'id-1',
        list: {
          'id-1': {
            id: 'id-1',
            label: null,
            metadata: [],
            requiredStatement: 'Some required statement',
            summary: null,
          },
        },
        order: ['id-1'],
      },
    });
  });
});
