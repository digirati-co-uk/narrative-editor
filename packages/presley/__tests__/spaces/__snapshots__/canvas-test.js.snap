// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spaces/canvas canvas can be created 1`] = `
Object {
  "canvas": Object {
    "current": "id-123",
    "list": Object {
      "id-123": Object {
        "id": "id-123",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
    },
    "order": Array [
      "id-123",
    ],
  },
}
`;

exports[`spaces/canvas canvas can be recovered 1`] = `
Object {
  "canvas": Object {
    "current": "id-3",
    "list": Object {
      "id-1": Object {
        "id": "id-1",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
      "id-2": Object {
        "id": "id-2",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
      "id-3": Object {
        "id": "id-3",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
    },
    "order": Array [
      "id-1",
      "id-3",
      "id-2",
    ],
  },
}
`;

exports[`spaces/canvas canvas can be removed 1`] = `
Object {
  "canvas": Object {
    "current": "id-3",
    "list": Object {
      "id-1": Object {
        "id": "id-1",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
      "id-2": Object {
        "id": "id-2",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
      "id-3": Object {
        "id": "id-3",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
    },
    "order": Array [
      "id-1",
      "id-3",
    ],
  },
}
`;

exports[`spaces/canvas canvas that does not exist cannot be recovered 1`] = `
Object {
  "canvas": Object {
    "current": null,
    "list": Object {
      "id-1": Object {
        "id": "id-1",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
    },
    "order": Array [],
  },
}
`;

exports[`spaces/canvas multiple canvases can be created 1`] = `
Object {
  "canvas": Object {
    "current": "id-2",
    "list": Object {
      "id-1": Object {
        "id": "id-1",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
      "id-2": Object {
        "id": "id-2",
        "label": null,
        "metadata": Array [],
        "requiredStatement": null,
        "summary": null,
      },
    },
    "order": Array [
      "id-1",
      "id-2",
    ],
  },
}
`;
